labgrid/driver/power/raritan.py

```diff
@@ -0,0 +1,60 @@
+from pysnmp import hlapi
+from ..exception import ExecutionError
+
+
+OID = ".1.3.6.1.4.1.13742.6.4.1.2.1"
+NUMBER_OF_OUTLETS = 16
+
+
+class _Snmp:
+    """A class that helps wrap pysnmp"""
+    def __init__(self, host, community, port=161):
+        if port is None:
+            port = 161
+
+        self.engine = hlapi.SnmpEngine()
+        self.transport = hlapi.UdpTransportTarget((host, port))
+        self.community = hlapi.CommunityData(community, mpModel=0)
+        self.context = hlapi.ContextData()
+
+    def get(self, oid):
+        g = hlapi.getCmd(self.engine, self.community, self.transport,
+            self.context, hlapi.ObjectType(hlapi.ObjectIdentity(oid)),
+            lookupMib=False)
+
+        error_indication, error_status, _, res = next(g)
+        if error_indication or error_status:
+            raise ExecutionError("Failed to get SNMP value.")
+        return res[0][1]
+
+    def set(self, oid, value):
+        identify = hlapi.ObjectType(hlapi.ObjectIdentity(oid),
+                   hlapi.Integer(value))
+        g = hlapi.setCmd(self.engine, self.community, self.transport,
+            self.context, identify, lookupMib=False)
+        next(g)
```

Comment:
This duplicates the code in `eaton.py`. Please move the class as `SimpleSNMP` to a newly created `labgrid/util/snmp.py` first. Then a second commit would use it from a the new raritan backend.